### PR TITLE
Move Thread PNSE check to managed to cut more dependencies

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

File: src/mono/netcore/System.Private.CoreLib/src/System/Threading/Thread.Browser.Mono.cs [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
+using System.Runtime.Versioning;
+
 namespace System.Threading
 {
     public partial class Thread
     {
         internal const bool IsThreadStartSupported = false;
+
+        [UnsupportedOSPlatform("browser")]
+        public void Start() => throw new PlatformNotSupportedException();
```
Comment (vargaz) [inline src/mono/netcore/System.Private.CoreLib/src/System/Threading/Thread.Browser.Mono.cs:13]: Note that threads might be supported in the browser in the future.

Comment (marek-safar) [reply src/mono/netcore/System.Private.CoreLib/src/System/Threading/Thread.Browser.Mono.cs:13]: This does not change the status-quo only make it more visible to managed side

**Review by vargaz — APPROVED** [review]

**Review by lewing — APPROVED** [review]

File: src/mono/netcore/System.Private.CoreLib/src/System/Threading/Thread.Browser.Mono.cs (hunk 1 above)
Comment (akoeplinger) [reply src/mono/netcore/System.Private.CoreLib/src/System/Threading/Thread.Browser.Mono.cs:13]: this should probably replicate the `"Cannot start threads on this runtime."` message that we had in native.

Comment (marek-safar) [reply src/mono/netcore/System.Private.CoreLib/src/System/Threading/Thread.Browser.Mono.cs:13]: We don't do it for other PNSE and I think in this case it'll be quite obvious from the stacktrace

**Pull request merged**